src/Lumi/Components2/QRCode.purs

```diff
@@ -1,67 +1,95 @@
 import React.Basic.Hooks as React
 import Web.DOM (Node)
+import Web.HTML.History (URL(..))
 
 newtype UseQRCode hooks
-  = UseQRCode (UseRef (Nullable.Nullable Node) (UseContext LumiTheme hooks))
+  = UseQRCode
+  ( UseEffect
+      (UnsafeReference (LumiComponent ()))
+      ( UseState
+          (Maybe URL)
+          ( UseMemo
+              (String /\ ErrorCorrectLevel)
+              (LumiComponent ())
+              (UseRef (Nullable.Nullable Node) hooks)
+          )
+      )
+  )
 
 derive instance ntUseQRCode :: Newtype (UseQRCode hooks) _
 
 data ErrorCorrectLevel
-  = L
-  | M
-  | Q
-  | H
+  = ECLLow
+  | ECLMedium
+  | ECLQuality
+  | ECLHigh
+
+derive instance eqErrorCorrectLevel :: Eq ErrorCorrectLevel
 
 errorCorrectLevelToString :: ErrorCorrectLevel -> String
 errorCorrectLevelToString = case _ of
-  L -> "L"
-  M -> "M"
-  Q -> "Q"
-  H -> "H"
+  ECLLow -> "L"
+  ECLMedium -> "M"
+  ECLQuality -> "Q"
+  ECLHigh -> "H"
 
-useQRCode :: String -> ErrorCorrectLevel -> Hook UseQRCode { qrcode :: JSX, saveOnClick :: String -> Effect Unit }
-useQRCode value level =
+useQRCode :: ErrorCorrectLevel -> String -> Hook UseQRCode { qrcode :: LumiComponent (), url :: Maybe URL }
+useQRCode level value =
   coerceHook React.do
-    theme <- useTheme
     ref <- useRef Nullable.null
-    pure
-      { qrcode:
-          E.element R.div'
-            { children:
-                [ element qrcode_
-                    { value
-                    , level: errorCorrectLevelToString level
-                    , renderAs: "svg"
-                    , xmlns: "http://www.w3.org/2000/svg"
-                    , size: 126
-                    }
-                ]
-            , ref
-            , className: ""
-            , css: toCSS theme { className: "", css: mempty } Styles.QRCode.qrcode
-            }
-      , saveOnClick: saveOnClick ref
-      }
+    qrcode <-
+      useMemo (value /\ level) \_ ->
+        unsafePerformEffect do
+          lumiComponent "QRCode" {} \props -> React.do
+            theme <- useTheme
+            pure
+              $ E.element R.div'
+                  { children:
+                      [ element qrcode_
+                          { value
+                          , level: errorCorrectLevelToString level
+                          , renderAs: "svg"
+                          , xmlns: "http://www.w3.org/2000/svg"
+                          , size: Nullable.null
+                          , bgColor: "rgba(255,255,255,0.0)"
+                          , fgColor: "rgba(0,0,0,1.0)"
+                          }
+                      ]
+                  , ref
+                  , className: props.className
+                  , css: toCSS theme props Styles.QRCode.qrcode
+                  }
+    url /\ setUrl <- useState Nothing
+    useEffect (UnsafeReference qrcode) do
```

Comment:
Just a suggestion/idea: I believe that using `useAff` here instead of `useEffect`+`useState` could simplify this.

Reply:
Or maybe it wouldn't that much? Because you'd have to use `makeAff` with an `effectCanceler` 🤔

Reply:
It probably wouldn't be a whole lot different. I'll leave as is for the sake of time.